lib/internal/streams/pipeline.js

```diff
@@ -147,13 +152,29 @@ async function pump(iterable, writable, finish) {
   }
   let error;
   try {
-    for await (const chunk of iterable) {
-      if (!writable.write(chunk)) {
-        if (writable.destroyed) return;
-        await EE.once(writable, 'drain');
+    if (!isStdio(writable)) {
+      for await (const chunk of iterable) {
+        if (!writable.write(chunk)) {
+          if (writable.destroyed) return;
+          await EE.once(writable, 'drain');
+        }
+      }
+      writable.end();
+    } else {
+      const errorPromise = new Promise((resolve, reject) => {
+        writable.on('error', reject);
+      });
+      for await (const chunk of iterable) {
+        await Promise.race([
+          errorPromise,
+          new Promise((resolve, reject) => {
+            writable.write(chunk, null, (err) => {
+              err ? reject(err) : resolve();
+            });
+          })
+        ]);
```

Comment:
It is likely faster to use the callbacks and manually handle everything. It could be wrapped in a single promise to allow to await the result for simplicity. Alternatively just also handle the `finish` call separately.

Reply:
> It is likely faster to use the callbacks and manually handle everything. It could be wrapped in a single promise to allow to await the result for simplicity

Yes, but that makes the implementation more complicated... at least as far as I can determine.

Reply:
@BridgeAR PTAL